tensorflow/python/kernel_tests/check_ops_test.py

```diff
@@ -1457,6 +1457,205 @@ def test_raises_when_wrong_type(self):
       check_ops.assert_type(floats, dtypes.float32)
 
 
+class AssertShapesTest(test.TestCase):
+
+  @test_util.run_in_graph_and_eager_modes
+  def test_raise_shape_mismatch(self):
+    x_known = array_ops.ones([3, 2], name="x")
+    y_known = array_ops.ones([2, 3], name="y")
+
+    shapes = {
+        x_known: ('N', 'Q'),
+        y_known: ('N', 'D'),
+    }
+    regex = r"Specified by tensor .* dimension 0.  " \
+            r"Tensor .* dimension 0 must have size 3.  Received size 2"
+    self.raises_static_error(shapes=shapes, regex=regex)
+
+    if not context.executing_eagerly():
```

Comment:
I think this non-eager second half is better written as its own separate test which explicitly enters graph mode with ops.Graph().as_default(). Same below.